driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -35,15 +40,26 @@ struct Context {
 impl UpdatingOrderbook {
     /// Does not block on initializing the orderbook. This will happen in the first call to
     /// `get_auction_data` which can thus take a long time to complete.
-    pub fn new(contract: Arc<dyn StableXContract + Send + Sync>, web3: Web3) -> Self {
+    pub fn new(
+        contract: Arc<dyn StableXContract + Send + Sync>,
+        web3: Web3,
+        path: Option<PathBuf>,
+    ) -> Self {
+        // Recover the orderbook from file (if possible, otherwise use default).
+        let orderbook = match &path {
+            Some(path) => Orderbook::try_from(path.as_path()).unwrap_or_default(),
+            None => Orderbook::default(),
+        };
+        let last_handled_block = orderbook.last_handled_block().unwrap_or(0);
```

Comment:
I would do this also as part of the initialize method implemented in #808 (since it could also be a little bit long running and is kind of optional)

Reply:
We can merge this first and I'll move it there in my PR.